tls/s2n_security_policies.c

```diff
@@ -980,6 +980,7 @@ const struct s2n_security_policy security_policy_rfc9151 = {
     .kem_preferences = &kem_preferences_null,
     .signature_preferences = &s2n_signature_preferences_rfc9151,
     .certificate_signature_preferences = &s2n_certificate_signature_preferences_rfc9151,
+    .certificate_preferences_apply_locally = true,
```

Comment:
Talked offline: let's separate this change out and do a little more investigation first.